resolved TODO - instrumentation names in suppressing doc (#2755)
open-telemetry/opentelemetry-java-instrumentation: docs/suppressing-instrumentation.md

## Suppressing specific agent instrumentation

You can suppress agent instrumentation of specific libraries by using
`-Dotel.instrumentation.[name].enabled=false` where `name` is the instrumentation `name`
(main or additional):

| Instrumentation name |
|---------------------------|
| akka-actor|
| akka-http|
| apache-camel|
| apache-dubbo|
| apache-httpasyncclient|
| apache-httpclient|
| armeria|
| async-http-client|
| aws-lambda|
| aws-sdk|
| axis2|
| cassandra|

> | classloader |

Reviewer: This module was recently renamed to `internal` (and it probably shouldn't be on this list)
```suggestion
```

Author: I see three names: `internal-class-loader`, `internal-proxy` and `internal-url-class-loader`. Why do you think these should not be on the list (ie user should not be able to disable them)?

Reviewer 2: Yes, they are required for agent, and should not be disabled.

Reviewer: Because without them the javaagent won't function properly (e.g. `helperResourceNames()` won't get injected).

Author: Sounds right :)

## Suppressing specific agent instrumentation

You can suppress agent instrumentation of specific libraries by using
`-Dotel.instrumentation.[name].enabled=false` where `name` is the instrumentation `name`
(main or additional):

| Main instrumentation name | Additional names |
|---------------------------|------------------|
| akka-actor|akka-actor-2.5 |
| akka-http|akka-http-client |
| akka-http|akka-http-server |
| apache-camel|apache-camel-2.20 |
| apache-dubbo|apache-dubbo-2.7 |
| apache-httpasyncclient|apache-httpasyncclient-4.1 |
| apache-httpclient|apache-httpclient-2.0 |
| apache-httpclient|apache-httpclient-4.0 |
| apache-httpclient|apache-httpclient-5.0 |
| armeria|armeria-1.3 |
| async-http-client|async-http-client-1.9 |
| async-http-client|async-http-client-2.0 |
| aws-lambda|aws-lambda-1.0 |
| aws-sdk|aws-sdk-1.11 |
| aws-sdk|aws-sdk-2.2 |
| axis2|axis2-1.6 |
| cassandra|cassandra-3.0 |
| cassandra|cassandra-4.0 |
| classloader |
| couchbase|couchbase-2.0 |
| couchbase|couchbase-2.6 |

> | couchbase|couchbase-3.1 |

Reviewer 3: I'd suggest that we swap these, so that the "Main instrumentation name" is unique and matches the instrumentation module name, while the additional names have the shortcuts / version-agnostic names

Author: I followed what we have in the code (main vs additional). It would definitely make sense to have the main as unique, but probably better to change it in the code and then reflect in the docs. What do you think?

Reviewer 3: 👍

```
public InstrumentationModule(
      String mainInstrumentationName, String... additionalInstrumentationNames) {
```

==>

```
public InstrumentationModule(String moduleName, String... additionalNames) {
```

where `moduleName` is the same as the gradle module name except when there are multiple InstrumentationModules in a single gradle module

Reviewer: > where `moduleName` is the same as the gradle module name except when there are multiple InstrumentationModules in a single gradle module

Hmm, if we follow this won't we end up with main module names like `couchbase-2.0`? In that case whenever somebody wants to turn off a particular instrumentation they'll need to know 1) what version of the instrumented library they're using, and 2) what instrumentations it maps to (taking couchbase as an example: version 2.8 actually uses both 2.0 and 2.6 instrumentations).
I think I'd rather get rid of those versioned module names completely (with several exceptions of course; for example the `couchbase-2.6` module could use names `"couchbase", "couchbase-networking"`, with every other couchbase module using just `"couchbase"`)

Author: That's a very interesting discussion (unique vs generic) but I think we should move it to a separate issue. Naming is important from both disabling and usage (checking in logs what actually happens) points of view. Deciding this here (documentation task) would not seem the right thing to do :)

Reviewer 3: I asked about this in the SIG meeting yesterday with @iNikem and @anuraaga, and consensus was to list the unversioned module names in this suppressing doc, and we can mention in troubleshooting about disabling a specific instrumentation module using the full module name, but that will just be a troubleshooting feature (e.g. for troubleshooting if muzzle is not doing its job) and we won't list all those versioned names in that doc. (and I think this means we don't need to change anything in the code)

Author: Updated :)

## Suppressing specific agent instrumentation

You can suppress agent instrumentation of specific libraries by using

> `-Dotel.instrumentation.[name].enabled=false` where `name` is the instrumentation `name`
> (main or additional):

Reviewer 3: ```suggestion
`-Dotel.instrumentation.[name].enabled=false` where `name` is the corresponding instrumentation name below:
```

> | Instrumentation name |
> |---------------------------|
> | akka-actor|

Reviewer 3: Single column table looks odd.

Maybe bullets?

Or maybe add "Library/Framework" column (matching https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/supported-libraries.md#libraries--frameworks), e.g.

```suggestion
| Library/Framework | Instrumentation name |
|-------------|--------------|
| Akka HTTP | akka-actor |
```

Reviewer 2: +1 for library name

You can suppress agent instrumentation of specific libraries by using
`-Dotel.instrumentation.[name].enabled=false` where `name` is the corresponding instrumentation `name`:

| Library/Framework | Instrumentation name |
|-------------------|----------------------|
| Akka Actor | akka-actor|
| Akka HTTP | akka-http|
| Apache Camel | apache-camel|
| Apache Dubbo | apache-dubbo|
| Apache HttpAsyncClient | apache-httpasyncclient|
| Apache HttpClient | apache-httpclient|
| Armeria | armeria|
| AsyncHttpClient (AHC) | async-http-client|
| AWS Lambda | aws-lambda|
| AWS SDK | aws-sdk|
| Apache Axis2 | axis2|
| Apache Cassandra | cassandra|
| Couchbase | couchbase|
| Apache CXF | cxf|
| Dropwizard Views | dropwizard-views |
| Eclipse OSGi | eclipse-osgi |
| Elasticsearch REST client | elasticsearch-rest|
| Elasticsearch client | elasticsearch-transport|
| java.util.concurrent | executor |
| Additional tracing annotations | external-annotations |
| Twitter Finatra | finatra|
| Apache Geode | geode|
| Google HTTP client | google-http-client|
| Grails | grails|
| Java EE Grizzly | grizzly|
| GRPC | grpc|
| Google Guava | guava|
| Google Web Toolkit | gwt|
| Hibernate | hibernate|

> | Java HTTP URL connection | http-url-connection |

Reviewer: Consider just using the class name:
```suggestion
| Java `HttpURLConnection` | http-url-connection |
```

Author: I find class name under "Library/Framework" confusing.

Reviewer 3: hm, tricky, I find the class name provides helpful specificity

| Java HTTP Client | httpclient |
| Netflix Hystrix | hystrix|
| JAX-RS Clieny | jaxrs-client|
| JAX-RS | jaxrs|
| JAX-WS | jaxws|
| Java JDBC | jdbc |

> | Java JDBC data source | jdbc-datasource |

Reviewer: ```suggestion
| Java JDBC `DataSource` | jdbc-datasource |
```

Author: I find class name under "Library/Framework" confusing.

Reviewer 3: same here

| Redis Jedis | jedis|
| Jetty | jetty|
| JMS | jms|
| JSP | jsp |
| Apache Kafka | kafka |
| Kotlin HTTP (kHttp) | khttp |
| kotlinx.coroutines | kotlinx-coroutines |
| K8s Client | kubernetes-client|
| Lettuce (Redis) | lettuce|
| OpenLiberty | liberty |
| Log4j | log4j|
| Logback | logback|
| Additional methods tracing | methods |
| JAX-WS Metro | metro|
| JSF Mojarra | mojarra|
| MongoDB | mongo |
| JSF MyFaces | myfaces|
| Netty | netty|
| OkHttp | okhttp|
| OpenTelemetry Trace annotations | opentelemetry-annotations |
| OSHI (Operating System and Hardware Information) | oshi |
| Play WS HTTP Client | play-ws|
| Play Framework | play|
| RabbitMQ Client | rabbitmq|
| Ratpack | ratpack|
| Reactor Netty | reactor-netty|
| Reactor | reactor|
| Rediscala | rediscala|
| Java RMI | rmi|
| Apache RocketMQ | rocketmq-client|
| ReactiveX RxJava | rxjava2 |
| Scala executors | scala-executors |
| Java Servlet | servlet|

> | Apache Spark | spark|

Reviewer: Actually it's not Apache Spark (big data processing), but [Spark Web Framework](http://sparkjava.com/)

```suggestion
| Spark Web Framework | spark|
```

| Elasticsearch client | elasticsearch-transport|
| Elasticsearch REST client | elasticsearch-rest|
| Google Guava | guava|
| Google HTTP client | google-http-client|
| Google Web Toolkit | gwt|
| Grails | grails|
| GRPC | grpc|
| Hibernate | hibernate|
| Java EE Grizzly | grizzly|
| Java HTTP Client | httpclient |
| Java `HttpURLConnection` | http-url-connection |
| Java JDBC | jdbc |
| Java JDBC `DataSource` | jdbc-datasource |
| Java RMI | rmi|
| Java Servlet | servlet|
| java.util.concurrent | executor |
| JAX-RS | jaxrs|
| JAX-RS Client | jaxrs-client|
| JAX-WS | jaxws|
| JAX-WS Metro | metro|
| Jetty | jetty|
| JMS | jms|
| JSF Mojarra | mojarra|
| JSF MyFaces | myfaces|
| JSP | jsp |
| K8s Client | kubernetes-client|
| Kotlin HTTP (kHttp) | khttp |
| kotlinx.coroutines | kotlinx-coroutines |
| Log4j | log4j|
| Logback | logback|
| MongoDB | mongo |
| Netflix Hystrix | hystrix|
| Netty | netty|
| OkHttp | okhttp|
| OpenLiberty | liberty |
| OpenTelemetry Trace annotations | opentelemetry-annotations |
| OSHI (Operating System and Hardware Information) | oshi |
| Play Framework | play|
| Play WS HTTP Client | play-ws|
| RabbitMQ Client | rabbitmq|
| Ratpack | ratpack|

> | ReactiveX RxJava | rxjava2 |

Reviewer: Oh, I almost forgot about it: recently RxJava3 instrumentation has been added:
```suggestion
| ReactiveX RxJava | rxjava2, rxjava3 |
```
